### PR TITLE
Add tooltips to #n log links

### PR DESCRIPTION
**Pull request by agrodet**
This PR adds a tooltip containing the target sentence text to `#n` links that appear in logs. It addresses #2293.

I factorized the code of the logs a little bit.  
I would also like to simplify the return statement of the following piece of code but I'm not sure how I should correct the corresponding tests
https://github.com/Tatoeba/tatoeba2/blob/70d6fc5200c9697d57f9f1db59096a3d78b44766/src/View/Helper/ClickableLinksHelper.php#L131-L138
Also, I'm not sure about how to add a note to translators so I tried to add it the same way I saw it somewhere else in the code.

PS: @jiru I didn't really understand your intention in your [comment about using view cells](https://github.com/Tatoeba/tatoeba2/issues/2293#issuecomment-627499720)

### CODE REVIEW

**Comment by jiru** [comment]
About the tests of that part of `ClickableLinksHelper.php`, they are actually in `tests/TestCase/View/Helper/MessagesHelperTest.php`. The ClickableLinks helper code is a bit different from the use case of your PR, because it needs to build a full URL (including the prefix https://tatoeba.org/...) because the link will show up in email notifications too. As opposed to that, your code doesn’t need the full links because it’s only used within pages. These two pieces of code could probably be factorized, but it’s important to understand this difference.

About the notes to translators, the way you did it is okay. If in doubt, you can try to run `./tools/generate_pot.sh` and see if your note show up in `src/Locale/default.pot`. I think the note can be either inside or outside the `__()`, as long as it is located on the line before the string itself.

About the view cells, I never used them so I’m not 100% sure, but basically, calling a model method from a view (using `TableRegistry::getTableLocator()` and so on) is a violation of the MVC principle, so we should avoid it as much as we can. But there are some specific cases when it’s sometimes needed, such as the clickable links. View cells give a proper way to achieve this. But it’s okay if you don’t want to go that far. :wink:

**Comment by jiru** [comment]
There is actually a way to get the sentence directly from the controller. You can modify `ContributionsTable::getContributionsRelatedToSentence()` so that it returns the sentence text of related sentences and translations using containments. Then, from the view, you can just get the related sentence text with `$log->sentence->text` and the related translation text with `$log->translation->text`. That would be a better design I think.

**Comment by jiru** [comment]
I did what I mentioned on my previous comment. Additionally, I made it so that the message "This sentence was initially added as a translation of sentence #n" also shows the sentence text as a tooltip.

File: src/View/Helper/ClickableLinksHelper.php [hunk 1]
```diff
@@ -183,16 +183,14 @@ public function buildSentenceLink($sentenceId, $sentenceText = null)
...
         return $this->Html->link(
             $linkText,
             array(
                 'controller' => 'sentences',
                 'action' => 'show',
                 $sentenceId
             ),
-            // $sentenceText could come as the empty string
-            array('title' => $sentenceText != null ? $sentenceText : $model->getSentenceTextForId($sentenceId))
+            ['title' => $sentenceText]
```
Comment (AndiPersti) [inline src/View/Helper/ClickableLinksHelper.php:193]: The sentence text needs to be sanitized using `$this->_View->safeForAngular()`.

BTW: Shouldn't we start using `md-tooltip` for all new tooltips? (see #1221)

Comment (jiru) [reply src/View/Helper/ClickableLinksHelper.php]: Good catch about the XSS! :sunglasses: 
I also updated the code to use `md-tooltip`.

**Pull request merged**